sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -858,7 +861,12 @@ class Analyzer(
         }
 
       case u: UnresolvedRelation =>
-        lookupRelation(u.multipartIdentifier).map(resolveViews).getOrElse(u)
+        val relationCache = AnalysisContext.get.relationCache
+        relationCache.getOrElse(u.tableName, {
```

Comment:
shall we apply the cache in `lookupRelation` where we expand the name and get the qualified table name?